### PR TITLE
Update cherry picks doc with a description for alpha features

### PR DESCRIPTION
**Pull request by dims**
We were doing this already, but wanted to make sure we write it down so it's easier for reviewers to point to something written down.

(nudged by discussion in https://github.com/kubernetes/kubernetes/pull/105011)

Signed-off-by: Davanum Srinivas <davanum@gmail.com>

**Which issue(s) this PR fixes**:

Fixes #

### CODE REVIEW

File: contributors/devel/sig-release/cherry-picks.md [hunk 1]
```diff
@@ -42,6 +42,11 @@ Again, the emphasis is on critical bug fixes, e.g.,
 - Panic, crash, hang
 - Security
 
+As a counter example, we would not support backporting a bug fix to an alpha
+feature in general. One condition where we would consider is if that bug caused
+an issue for in another part of the system even when the alpha feature was
+switched off / not in use.
+
```
Comment (liggitt) [inline contributors/devel/sig-release/cherry-picks.md:49]: sorry for the post-merge comments, a few nits...

* consider avoiding "we" language
* A legitimately critical issue like data loss or a security issue could still potentially be a reason for a backport for something even in the alpha state, depending on risk to the rest of the system

Would language like this map more closely to our current backport standards?

> A bugfix for a functional issue (not a data loss or security issue) that only affects an alpha feature does not qualify as a critical bug fix.

Comment (dims) [reply contributors/devel/sig-release/cherry-picks.md:49]: @liggitt please go ahead and open another PR. +1 from me

Comment (aojea) [reply contributors/devel/sig-release/cherry-picks.md:49]: +1

**Pull request merged**